src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Char.cs

```diff
@@ -687,11 +689,11 @@ public static unsafe int IndexOf(ref char searchSpace, char value, int length)
                         {
                             Debug.Assert(lengthToExamine >= Vector128<ushort>.Count);
 
-                            Vector128<ushort> search = LoadVector128(ref searchSpace, offset);
+                            Vector128<ushort> search = Vector128.LoadUnsafe(ref ushortSearchSpace, (uint)offset);
 
                             // Same method as above
-                            int matches = Sse2.MoveMask(Sse2.CompareEqual(values, search).AsByte());
-                            if (matches == 0)
+                            Vector128<ushort> compareResult = Vector128.Equals(values, search);
```

Comment:
Same question as on the `byte` PR.

What is the cost of this approach vs doing it every loop for Vector256<T>

Is it better, particularly for large inputs, to do this there as well?

Reply:
@tannergooding is slightly worse compared to the PR (4%):

```diff
- uint matches = Vector256.Equals(values, search).AsByte().ExtractMostSignificantBits();
- if (matches == 0)
+ Vector256<ushort> compareResult = Vector256.Equals(values, search);
+ if (compareResult == Vector256<ushort>.Zero)
  {
      // Zero flags set so no matches
      offset += Vector256<ushort>.Count;
      lengthToExamine -= Vector128<ushort>.Count;
      continue;
  }

+ uint matches = compareResult.AsByte().ExtractMostSignificantBits();
```

```ini
BenchmarkDotNet=v0.13.1.1828-nightly, OS=Windows 10 (10.0.18363.2212/1909/November2019Update/19H2)
Intel Xeon CPU E5-1650 v4 3.60GHz, 1 CPU, 12 logical and 6 physical cores
.NET SDK=7.0.100-rc.1.22405.1
  [Host]     : .NET 7.0.0 (7.0.22.40308), X64 RyuJIT AVX2
  Job-ZEKBPR : .NET 7.0.0 (42.42.42.42424), X64 RyuJIT AVX2
  Job-XEENUE : .NET 7.0.0 (42.42.42.42424), X64 RyuJIT AVX2
  Job-FOODNB : .NET 7.0.0 (42.42.42.42424), X64 RyuJIT AVX2

LaunchCount=9
```

|       Method |             Toolchain | Size |     Mean | Ratio |
|------------- |---------------------- |----- |---------:|------:|
| IndexOfValue | \question\corerun.exe |  512 | 13.09 ns |  0.99 |
| IndexOfValue |     \main\corerun.exe |  512 | 13.18 ns |  1.00 |
| IndexOfValue |       \PR\corerun.exe |  512 | 12.48 ns |  0.95 |

<img width="1768" alt="vtune" src="https://user-images.githubusercontent.com/6011991/183081148-5da2230d-30ef-414f-b269-d1a63f8c8312.png">